game-core/src/system/camera.rs

```diff
@@ -0,0 +1,48 @@
+use amethyst::renderer::Camera; 
+use amethyst::{
+    core::Transform,
+    ecs::{Join, ReadStorage, System, WriteStorage},
+};
+
+use crate::component::Player;
+
+pub struct Movement;
+
+impl<'s> System<'s> for Movement {
+    type SystemData = (
+        ReadStorage<'s, Player>,
+        ReadStorage<'s, Camera>,
+        WriteStorage<'s, Transform>,
+    );
+
+    fn run(&mut self, (players, cameras, mut transforms): Self::SystemData) {
+        let mut player_x: f32 = 0.0;
+        let mut player_y: f32 = 0.0;
```

Comment:
You can use tuple declarations to shorten this, so it could be rewritten as

```rust
let (mut player_x, mut player_y) = (0.0, 0.0);
```

Side note I dont think the type annotations here are necessary. I may be wrong.